### PR TITLE
*: abstract RangeProperties

### PR DESCRIPTION
**Pull request by Rustin-Liu**
Signed-off-by: Rustin-Liu <rustin.liu@gmail.com>

###  What have you changed?
abstract RangeProperties

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

cargo test --all --no-run

###  Refer to a related PR or issue link (optional)
part of https://github.com/tikv/tikv/pull/6739#issuecomment-593128992

### CODE REVIEW

**Comment by Rustin-Liu** [comment]
@zhangjinpeng1987  @brson  @hicqu  PTAL~

File: components/raftstore/src/store/fsm/store.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use engine::rocks;
 use engine::DB;
 use engine_rocks::{Compat, RocksCompactionJobInfo, RocksEngine, RocksWriteBatch};
 use engine_traits::{
-    CompactionJobInfo, KvEngine, Mutable, WriteBatch, WriteBatchExt, WriteOptions,
+    CompactionJobInfo, KvEngine, Mutable, RangeProperties, WriteBatch, WriteBatchExt, WriteOptions,
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/fsm/store.rs]: RocksRangeProperties?

Comment (Rustin-Liu) [reply components/raftstore/src/store/fsm/store.rs]: https://github.com/tikv/tikv/blob/589888a1ba55b895895adf93026efd8504e90144/components/raftstore/src/store/fsm/store.rs#L2074

**Comment by Rustin-Liu** [comment]
@zhangjinpeng1987  @brson  ping~

**Comment by Rustin-Liu** [comment]
@zhangjinpeng1987 @brson @hicqu  ping~

**Pull request closed without merging**